mk-local-include.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+# Binutils provide certain headers that are clashing with system-wide
+# headers on Cygwin and MacOS. They do so in different manners, though,
+# so it is not possible to just tweak the inclusion/library order.
+# Hence, we prepare a filtered inclusion directory where we only place
+# the headers that we need for BFD/binutils. To determine them, we'll
```

Comment:
the dynamic angle makes things a bit more robust to changes, but then we have to deal with parsing the output of the preprocessor which has its own issues (especially from a portability across systems angle) as can be seen from the work you had to put into this script :).

since the required binutils header list has been fairly stable over time (measured in years), i think it's OK to just hardcode the list like we already do.